### PR TITLE
Support More Web3 Providers

### PR DESCRIPTION
**Pull request by raghava-pamula**
Initial Commit for adding support for web3 providers outside of infura. Currently, the project only supports infura web3 providers, and more flexibility is needed since infura has had outages.

https://github.com/Uniswap/routing-api/issues/38

### CODE REVIEW

File: lib/handlers/injector-sor.ts [hunk 1]
```diff
@@ -108,9 +108,20 @@ export abstract class InjectorSOR<Router, QueryParams> extends Injector<
...
-        const url = `https://${chainName}.infura.io/v3/${projectId}`
+        let url = ''
+        switch (providerName) {
+          case 'infura':
+            url = `https://${chainName}.infura.io/v3/${projectId}`
+            break
+          case 'alchemy':
+          // todo: add logic to generate alchemy rpc urls
+          case 'moralis':
+          // todo: todo: add logic to generate moralis rpc urls
```
Comment (willpote) [inline lib/handlers/injector-sor.ts:121]: Could you fill in alchemy just so its ready if we need to switch out?

Think we can remove moralis too for now

**Review by willpote — COMMENTED** [review]
Looks good. One minor comment, and then just let me know when you are able to get the integ tests passing then I'll approve

**Comment by raghava-pamula** [comment]
Integ tests are failing :'(

File: bin/app.ts [hunk 2]
```diff
@@ -116,17 +120,24 @@ export class RoutingAPIPipeline extends Stack {
       secretCompleteArn: 'arn:aws:secretsmanager:us-east-2:644039819003:secret:hosted-zone-JmPDNV',
     })
 
+    // Parse AWS Secret
+    let jsonRpcProviders = {} as { [chainId: string]: string }
+    SUPPORTED_CHAINS.forEach((chainId: ChainId) => {
+      const key = `WEB3_RPC_${chainId}`
+      jsonRpcProviders[key] = jsonRpcProvidersSecret.secretValueFromJson(key) as unknown as string
```
Comment (willpote) [inline bin/app.ts:127]: Use `.toString()` like we do below

File: bin/app.ts [hunk 3]
```diff
@@ -204,7 +215,19 @@ const app = new cdk.App()
...
+    WEB3_RPC_4: process.env.WEB3_RPC_4!,
+    WEB3_RPC_5: process.env.WEB3_RPC_5!,
+    WEB3_RPC_42: process.env.WEB3_RPC_42!,
+    WEB3_RPC_10: process.env.WEB3_RPC_10!,
+    WEB3_RPC_69: process.env.WEB3_RPC_69!,
+    WEB3_RPC_42161: process.env.WEB3_RPC_42161!,
+    WEB3_RPC_421611: process.env.WEB3_RPC_421611!,
+    WEB3_RPC_137: process.env.WEB3_RPC_137!,
+    WEB3_RPC_80001: process.env.WEB3_RPC_80001!,
+  },
```
Comment (willpote) [inline bin/app.ts:230]: Can you update the package README so people know to do this?

File: lib/handlers/injector-sor.ts [hunk 4]
```diff
@@ -107,10 +106,12 @@ export abstract class InjectorSOR<Router, QueryParams> extends Injector<
 
     const dependenciesByChainArray = await Promise.all(
       _.map(SUPPORTED_CHAINS, async (chainId: ChainId) => {
-        const chainName = ID_TO_NETWORK_NAME(chainId)
-        // updated chainNames to match infura strings
-        const projectId = process.env.PROJECT_ID
-        const url = `https://${chainName}.infura.io/v3/${projectId}`
+        const url = process.env[`WEB3_RPC_${chainId.toString()}`]!
+        log.info({ url: url }, `GENERATED URL`)
```
Comment (willpote) [inline lib/handlers/injector-sor.ts:110]: Lets remove this log

File: lib/handlers/injector-sor.ts [hunk 5]
```diff
@@ -107,10 +106,12 @@ export abstract class InjectorSOR<Router, QueryParams> extends Injector<
...
+
+        if (typeof url == undefined || url == '') {
```
Comment (willpote) [inline lib/handlers/injector-sor.ts:112]: Just `!url` is fine

File: bin/app.ts [hunk 6]
```diff
@@ -93,8 +95,10 @@ export class RoutingAPIPipeline extends Stack {
     // Secrets are stored in secrets manager in the pipeline account. Accounts we deploy to
     // have been granted permissions to access secrets via resource policies.
 
-    const infuraProjectId = sm.Secret.fromSecretAttributes(this, 'InfuraProjectId', {
-      secretCompleteArn: 'arn:aws:secretsmanager:us-east-2:644039819003:secret:infuraProjectId-UlSwK2',
+    const jsonRpcProvidersSecret = sm.Secret.fromSecretAttributes(this, 'RPCProviderUrls', {
+      secretCompleteArn:
+        'arn:aws:secretsmanager:us-east-2:644039819003:secret:routing-api-rpc-urls-json-primary-ixS8mw',
+      //backup Arn: arn:aws:secretsmanager:us-east-2:644039819003:secret:routing-api-rpc-urls-json-backup-D2sWoe
```
Comment (willpote) [inline bin/app.ts:101]: Could you add a bit more context to this comment so other people know why this exists? 

Also add a note that you must update the multicall chunk size to 50  for optimism when switching to the backup

**Review by willpote — CHANGES_REQUESTED** [review]

**Review by willpote — APPROVED** [review]

**Pull request merged**